modules/social_features/social_user/social_user.module

```diff
@@ -144,16 +143,25 @@ function social_user_form_views_exposed_form_alter(&$form, FormStateInterface $f
 /**
  * Returns array with titles of all groups, ordered by their label.
  */
-function _social_user_get_groups() {
-  $data = &drupal_static(__FUNCTION__);
+function _social_user_get_groups(): array {
+  if (!empty($data = &drupal_static(__FUNCTION__, []))) {
+    return $data;
+  }
+
+  $groups = \Drupal::database()->select('groups_field_data', 'gfd')
+    ->fields('gfd', ['id', 'label', 'type'])
+    ->orderBy('type')
+    ->orderBy('label')
+    ->execute();
+
+  if ($groups === NULL || !($groups = $groups->fetchAll())) {
+    return $data;
+  }
 
-  if (empty($data)) {
-    $data = \Drupal::database()
-      ->select('groups_field_data', 'gfd')
-      ->fields('gfd', ['id', 'label'])
-      ->orderBy('label')
-      ->execute()
-      ->fetchAllKeyed(0, 1);
+  $bundles = \Drupal::service('entity_type.bundle.info')->getBundleInfo('group');
+
+  foreach ($groups as $group) {
+    $data[$bundles[$group->type]['label']][$group->id] = $group->label;
```

Comment:
It seems like we are restructuring the data returned by this function now including group type.
This means that the
` * Returns array with titles of all groups, ordered by their label.` function description also doesn't match anymore.
And also means it's not backwards compatible anymore

Reply:
I've fixed the description of the function. Also, I've looked for use cases of this function in Distro and enterprise extensions and found only one place in Distro where it using. Do you know other places where this function is used? If not, then doesn't make sense to think about backward compatibility in my opinion.

Reply:
We need to think about our community as well that might use this ;)

Reply:
Good reason. Fixed.